Intersect.Client.Core/Utilities/TextColorParser.cs

```diff
@@ -5,45 +5,23 @@ namespace Intersect.Client.Utilities
 {
     public static class TextColorParser
     {
-        private static readonly Dictionary<string, Func<Color>> KnownColors = InitializeColorDictionary();
-        private static readonly Regex ColorTagRegex = new(@"\\c{([^}]*)}", RegexOptions.Compiled);
-        private static readonly string SplitPattern = @"(\\c{[^}]*})";
-
-        private static Dictionary<string, Func<Color>> InitializeColorDictionary()
-        {
-            return typeof(Color)
-                .GetProperties(BindingFlags.Public | BindingFlags.Static)
-                .Where(p => p.PropertyType == typeof(Color))
-                .ToDictionary(
-                    p => p.Name.ToLowerInvariant(),
-                    p => (Func<Color>)(() => p.GetMethod?.Invoke(null, null) as Color ?? Color.White)
-                );
-        }
+        private static readonly Regex ColorTagRegex = new(@"(\\c{[^}]*})", RegexOptions.Compiled);
 
         public static List<ColorizedText> Parse(string text, Color defaultColor)
         {
-            var output = new List<ColorizedText>();
-            var segments = Regex.Split(text, SplitPattern)
-                .Where(s => !string.IsNullOrWhiteSpace(s));
+            var segments = ColorTagRegex.Split(text)
+                .Where(s => !string.IsNullOrWhiteSpace(s))
+                .ToArray();
 
+            var output = new List<ColorizedText>(segments.Length);
             var currentColor = defaultColor;
 
             foreach (var segment in segments)
             {
-                var match = ColorTagRegex.Match(segment);
-
-                if (match.Success)
+                if (segment.StartsWith("\\c{") && segment.EndsWith("}"))
                 {
-                    string colorCode = match.Groups[1].Value.ToLowerInvariant();
-
-                    if (KnownColors.TryGetValue(colorCode, out Func<Color>? colorFunc))
-                    {
-                        currentColor = colorFunc();
-                    }
-                    else
-                    {
-                        currentColor = Color.FromHex(colorCode, defaultColor) ?? defaultColor;
-                    }
+                    string colorCode = segment[3..^1].ToLowerInvariant();
+                    currentColor = Color.FromString(colorCode, defaultColor) ?? defaultColor;
```

Comment:
`?? defaultColor` isn't necessary, if it returns null from `FromString()` then `defaultColor` already has to be null.